c7n/resources/route53.py

```diff
@@ -194,6 +194,39 @@ def process_resource_set(self, client, domains, keys):
                 TagsToDelete=keys)
 
 
+@HostedZone.action_registry.register('delete')
+class Delete(BaseAction):
+    """Action to delete Route 53 hosted zones
+
+    It is recommended to use a filter to avoid unwanted deletion of R53 hosted zones
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: route53-delete-testing-hosted-zones
+                resource: aws.hostedzone
+                filters:
+                  - 'tag:TestTag': present
+                actions:
+                  - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('route53:DeleteHostedZone')
+
+    def process(self, hosted_zones):
+        with self.executor_factory(max_workers=3) as w:
```

Comment:
we can probably ditch the concurrency here, we'll get hit by rate limits deleting zones, and generally the cardinality is not extreme afaik.

Reply:
addressed